src/wcs2_0/src/main/java/org/geoserver/wcs2_0/GetCoverage.java

```diff
@@ -1605,7 +1605,7 @@ private Map<String, InterpolationPolicy> extractInterpolation(
             CoverageInfo ci, Map<String, ExtensionItemType> extensions) {
         // preparation
         final Map<String, InterpolationPolicy> returnValue = new HashMap<>();
-        final Envelope envelope = ci.getNativeBoundingBox();
+        final Envelope envelope = WCSUtils.toNativeBounds(ci);
```

Comment:
By adding some try/catch statements ``ci.boundingBox()`` can also be used here; however it is willing to make a lot more assumptions to produce a native bounding box.

WCSUtils.toNativeBounds(ci) is more of a minimal drop-in replacement.